### PR TITLE
updates README to refer 0.3.0 release

### PR DESCRIPTION
**Pull request by hrishin**
# Changes
updates README to refer 0.3.0 release

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```
release-note
```

### CODE REVIEW

**Comment by chmouel** [comment]
I think we should just update it to point to /latest/

**Comment by hrishin** [comment]
Yes @chmouel every time updating it doesn't make sense. However, wondering sure about changing to the latest would solve that?

**Comment by vdemeester** [comment]
@chmouel the problem is that the tarball generated have the version in their name, so the link would be wrong with `latest` :sweat_smile:

**Comment by chmouel** [comment]
so options :

1) tell people to grab the tarball for their operating system in : 

https://github.com/tektoncd/cli/releases/latest

and no need to list/update a bunch of links

2) give people something like this : 

```
% latest=$(curl -s https://api.github.com/repos/tektoncd/cli/releases/latest|python -c "import sys, json;x=json.load(sys.stdin);print(x['tag_name'].replace('v',''))");echo https://github.com/tektoncd/cli/releases/download/v${latest}/tkn_${latest}_Darwin_x86_64.tar.gz
```

3) update every time the readme manually and die of boreness 4eva

File: README.md [hunk 1]
```diff
@@ -11,13 +11,13 @@ Tekton!
 
 Download the latest binary executable for your operating system:
 
-* [Mac OS X](https://github.com/tektoncd/cli/releases/download/v0.2.2/tkn_0.2.2_Darwin_x86_64.tar.gz)
```
Comment (sthaha) [inline README.md]: wonder if this can be replaced with `latest` instead of a version number?

Comment (chmouel) [reply README.md]: :D https://github.com/tektoncd/cli/pull/267#issuecomment-527424693

**Pull request merged**